### DISCUSSION TITLE
Replace monkey patching?

### DISCUSSION THREAD

**Discussion by willmcgugan** [ewels/rich-click#7] [Ideas]
Would it be possible to apply the Rich formatting without monkey patching?

I was thinking that the `rich_click` namespace could replicate the `click` namespace. But supply customized version of Click objects (via inheritance).

This is what we have currently:

```python
import click
import rich_click

class RichClickGroup(click.Group):
    def format_help(self, ctx, formatter):
        rich_click.rich_format_help(self, ctx, formatter)
class RichClickCommand(click.Command):
    def format_help(self, ctx, formatter):
        rich_click.rich_format_help(self, ctx, formatter)

@click.group(cls=RichClickGroup)
@click.option('--debug/--no-debug', default=False)
def cli(debug):
    click.echo(f"Debug mode is {'on' if debug else 'off'}")

@cli.command(cls=RichClickCommand)
def sync():
    click.echo('Syncing')
```

But I think it could be as simple (for the user at least) as this:

```python
import rich_click as click 

@click.group()
@click.option('--debug/--no-debug', default=False)
def cli(debug):
    click.echo(f"Debug mode is {'on' if debug else 'off'}")

@cli.command()
def sync():
    click.echo('Syncing')
```

Thoughts?

**Comment by ewels** [comment]
Beautiful! Even _less_ typing to get it working, I didn't think that was possible.. 😄 And yes this seems cleaner than the monkey patching approach.. 👍🏻

**Reply by ewels** [reply]
Done in PR #10 - thanks for the suggestion! 🚀
